### PR TITLE
Adding ReadOnlyDatabase object to Synapse model.

### PR DESCRIPTION
**Pull request by amirdach**
### Changelog

This PR updates an existing API version and adds a missing object.  
It adds ReadOnlyDatabase (and its properties object) to the Synapse model which is similar to the ReadOnlyDatabase in Kusto model.

### CODE REVIEW

**Review by docohe — APPROVED** [review]
Could you change the readwrite database statistics to readOnly:true?
Should be done in 2947

**Review by wonner — APPROVED** [review]

**Review by ruowan — APPROVED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json [hunk 1]
```diff
@@ -6122,8 +6122,7 @@
         "storageAccountSubscriptionId": {
           "format": "uuid",
           "description": "Specifies the blob storage subscription Id.",
-          "type": "string",
-          "default": ""
```
Comment (msyyc) [inline specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/sqlPool.json:6126]: Fix python pipeline failure

**Review by ruowan — APPROVED** [review]

**Pull request merged**